src/validation/cbor.rs

```diff
@@ -675,14 +696,32 @@ impl Validator<Value> for CDDL {
           unimplemented!()
         }
       }
-      GroupEntry::TypeGroupname { ge: tge, .. } => self.validate_rule_for_ident(
-        &tge.name,
-        is_enumeration,
-        None,
-        None,
-        tge.occur.as_ref(),
-        value,
-      ),
+      GroupEntry::TypeGroupname { ge: tge, span } => {
+        if is_type_prelude(&tge.name.ident) {
+          // Substitute a new AST node for the groupentry validation.
+          // FIXME: this seems like an awkward thing to do.
+          self.validate_type2(
+            &Type2::Typename {
+              ident: tge.name.clone(),
+              generic_arg: tge.generic_arg.clone(),
+              span: span.clone(),
```

Comment:
```suggestion
              span,
```

Reply:
This makes the compiler sad:
```
expected tuple, found reference
help: consider dereferencing the borrow: `span: *span`
```
Span is Copy, so I think that `*span` and `span.clone()` are effectively the same thing.  (And I'm happy to follow whatever style you prefer)

If in the future Span might become a nontrivial non-Copy type, I think `*span` is likely to trigger "move out of borrowed context" errors.

Reply:
Ah yep. Good catch. Can leave as is then.